File: 80. Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() < 3) return nums.size();
        int res = 1, left = 0, right = 1, cnt = 0;
        while (right < nums.size()) {
            if (nums[right] != nums[left] || cnt == 0) {
                if (nums[right] == nums[left]) {
                    cnt++;
                } else {
                    cnt = 0;
                }
                nums[++left] = nums[right++];
                res++;
            } else {
                while (right < nums.size() && nums[right] == nums[left]) {
                    right++;
                }
            }
        }
        return res;
    }
};

File: java/80. Remove Duplicates from Sorted Array II.java
// O(n)
class Solution {
    public int removeDuplicates(int[] nums) {
        int i = 0, j = 0, count = 0;
        while (j < nums.length) {
            if (j == 0 || nums[j] != nums[j - 1]) {
                nums[i++] = nums[j];
                count = 1;
            } else if (count < 2) {
                nums[i++] = nums[j];
                ++count;
            }
            ++j;
        }
        return i;
    }
}
